internal/services/sentinel/sentinel_alert_rule_fusion_resource.go

```diff
@@ -158,25 +162,69 @@ func resourceSentinelAlertRuleFusionCreateUpdate(d *pluginsdk.ResourceData, meta
 		},
 	}
 
-	if !d.IsNewResource() {
-		resp, err := client.Get(ctx, id)
-		if err != nil {
-			return fmt.Errorf("retrieving %q: %+v", id, err)
+	if _, err := client.CreateOrUpdate(ctx, id, params); err != nil {
+		return fmt.Errorf("creating Sentinel Alert Rule Fusion %q: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceSentinelAlertRuleFusionRead(d, meta)
+}
+
+func resourceSentinelAlertRuleFusionUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Sentinel.AlertRulesClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := alertrules.ParseAlertRuleID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, *id)
+	if err != nil {
+		return fmt.Errorf("retrieving %q: %+v", id, err)
+	}
+
+	if err = assertAlertRuleKind(resp.Model, alertrules.AlertRuleKindFusion); err != nil {
+		return fmt.Errorf("asserting alert rule of %q: %+v", id, err)
+	}
+
+	payload := resp.Model.(alertrules.FusionAlertRule)
+
+	if d.HasChange("alert_rule_template_guid") {
+		if payload.Properties == nil {
+			payload.Properties = &alertrules.FusionAlertRuleProperties{}
 		}
+		payload.Properties.AlertRuleTemplateName = d.Get("alert_rule_template_guid").(string)
+	}
 
-		if resp.Model == nil {
-			return fmt.Errorf("retrieving %q: model was nil", id)
+	if d.HasChange("enabled") {
+		if payload.Properties == nil {
+			payload.Properties = &alertrules.FusionAlertRuleProperties{}
 		}
-		if err = assertAlertRuleKind(resp.Model, alertrules.AlertRuleKindFusion); err != nil {
-			return fmt.Errorf("asserting alert rule of %q: %+v", id, err)
+		payload.Properties.Enabled = d.Get("enabled").(bool)
+	}
+
+	if d.HasChange("source") {
+		if payload.Properties == nil {
+			payload.Properties = &alertrules.FusionAlertRuleProperties{}
 		}
+		payload.Properties.SourceSettings = expandFusionSourceSettings(d.Get("source").([]interface{}))
 	}
 
-	if _, err := client.CreateOrUpdate(ctx, id, params); err != nil {
-		return fmt.Errorf("creating Sentinel Alert Rule Fusion %q: %+v", id, err)
+	// The `Description` is read-only but not specified on the Swagger, tracked on: https://github.com/Azure/azure-rest-api-specs/issues/31330
+	if payload.Properties != nil {
+		payload.Properties.Description = nil
+		payload.Properties.DisplayName = nil
+		payload.Properties.LastModifiedUtc = nil
+		payload.Properties.Severity = nil
+		payload.Properties.Tactics = nil
 	}
```

Comment:
We don't need to nil check this anymore since we're doing it further up
```suggestion
	payload.Properties.Description = nil
	payload.Properties.DisplayName = nil
	payload.Properties.LastModifiedUtc = nil
	payload.Properties.Severity = nil
	payload.Properties.Tactics = nil
```